translator.py

```diff
@@ -236,59 +255,83 @@ def _alloc_mcast_addr(self, client_addr):
-            # to) as otherwise we'll get rejected by the server.
-            form_params = {'csrfmiddlewaretoken': csrftoken, 'source': self._mcast_src_ip, 'group': str(mcast_dst_ip),
-                           'udp_port': str(self._mcast_dst_port), 'email': email, 'description': str(description),
-                           'Add': 'Add'}
-            header_fields = {'Referer': constants.MULTICASTMENU_ADD_URL}
-            resp_post = sess.post(constants.MULTICASTMENU_ADD_URL, data=form_params, headers=header_fields)
-            if resp_post.status_code != 200:
+            options = {
+                "unique_identifier": self._mm_uid,
+                "source": self._mcast_src_ip,
+                "group": str(mcast_dst_ip),
+            }
+
+            if src_ip[0] == constants.MULTICASTMENU_SENDER_IP:
+                print("'Inside Request' set to True")
+                options["inside_request"] = "1"
+
+            resp_post = requests.post(constants.MULTICASTMENU_ADD_URL, data=options)
+
+            if resp_post.status_code != 201:
                 errmsg += f'POST resulted in status code {resp_post.status_code}. Response body:\n{resp_post.text}'
                 print(errmsg)
                 return
+
+            access_code = json.loads(resp_post.text)["data"].split(" ")[-1]
+            self._acm[src_ip] = access_code
+
             msg = f'Added amt://{self._mcast_src_ip}@{mcast_dst_ip}:{self._mcast_dst_port} to the Multicast Menu ' \
-                  f'(email={email}; description={description}).'
+                  f'(source={src_ip}). The access code is {access_code}.'
             print(msg)
+
         except Exception as e:
             errmsg += f'encountered a {type(e)} with message "{e}".'
             print(errmsg)
 
-    # TODO add functionality that evicts addresses from the FIB when they've been inactive for a while.
-
+    def _identify_inactive_streams(self):
+        """
+        Loop through the streams in the FIB and determine which have not been seen in a while.
+        """
+        try:
+            self._lock.acquire()
+            to_remove_from_fib = []
+
+            time = datetime.now()
+            for src_addr in self._fib:
+                if time - self._atb[src_addr] > timedelta(seconds=20):
+                    # Remove this stream from the FIB
+                    self._allocated_mcast_addrs.remove(self._fib[src_addr])
+
+                    if self._mm_uid is not None:
+                        # Remove the Multicast Menu entry for this stream
+                        options = {
+                            "unique_identifier": self._mm_uid,
+                            "access_code": self._acm[src_addr],
+                        }
+
+                        resp_post = requests.post(constants.MULTICASTMENU_REMOVE_URL, data=options)
+
+                        if resp_post.status_code != 201:
```

Comment:
HTTP status code 201 indicates resource creation, but we're doing the opposite here (deletion). Don't change this here if the API currently returns 201 for this operation - just a heads up in case it's a typo here.